lib/EarlyFunctionAnalysis/UsedRegisters.h

```diff
@@ -0,0 +1,152 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <map>
+#include <set>
+
+#include "revng/EarlyFunctionAnalysis/CallGraph.h"
+#include "revng/EarlyFunctionAnalysis/Common.h"
+#include "revng/MFP/SetLattices.h"
+#include "revng/Model/Register.h"
+#include "revng/Support/MetaAddress.h"
+
+#include "DetectABI.h"
+
+namespace efa {
+
+struct FunctionABI {
+  using SetOfRegisters = std::set<model::Register::Values>;
+  using SUL = SetUnionLattice<SetOfRegisters>;
+  SetOfRegisters ArgumentRegisters;
+  SetOfRegisters ReturnRegisters;
+};
+
+// using LatticeElement = std::map<MetaAddress, FunctionABI>;
+
+struct LatticeElement : std::map<MetaAddress, FunctionABI> {
```

Comment:
I'd avoid inheritance here, better to use composition (i.e., have a `std::map` field) and expose only the methods you need.